src/comments.js

```diff
@@ -381,6 +383,20 @@ function handleConditionalExpressionComments(enclosingNode, followingNode, comme
   return false;
 }
 
+function handleAssignmentPatternComments(enclosingNode, comment) {
```

Comment:
It's not the wrong approach if we have to do it, but I'd also like to know why we have to. If like you say it triggers that branch and attaches it as a trailing comment to the identifier, it should be printed.

Search the code for `comment.printed`; it's only ever set the `true` and `false.` Nothing ever branched on it, so can you look into why something is changed and why the comment isn't printed in the first place?